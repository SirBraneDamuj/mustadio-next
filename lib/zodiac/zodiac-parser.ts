import { Zodiacs } from "../types";

function parseZodiacLine(zodiacLine: string) {
  const [name, info] = zodiacLine.split(" compatibility: ");
  return {
    name,
    info,
  };
}

export function parseZodiacs(zodiacsDump: string): Zodiacs {
  const zodiacs: Zodiacs = {};
  const zodiacLines = zodiacsDump.trim().split("\n");

  for (const zodiacLine of zodiacLines) {
    if (zodiacLine.trim() === "") continue; // Skip empty lines
    const zodiac = parseZodiacLine(zodiacLine);
    zodiacs[zodiac.name] = zodiac;
  }

  return zodiacs;
}
