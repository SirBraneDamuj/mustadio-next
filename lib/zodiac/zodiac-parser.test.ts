import { promises as fs } from "fs";
import path from "path";
import { describe, expect, it } from "vitest";
import { parseZodiacs } from "./zodiac-parser";

describe("parseZodiacs", () => {
  it("parses zodiachelp.txt correctly (snapshot)", async () => {
    const filePath = path.join(
      __dirname,
      "../../resources/fftbg_fake/zodiachelp.txt"
    );
    const file = await fs.readFile(filePath, "utf-8");
    const result = parseZodiacs(file);

    expect(result).toMatchSnapshot();
  });
});
