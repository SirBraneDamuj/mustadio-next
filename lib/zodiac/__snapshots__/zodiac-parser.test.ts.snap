// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseZodiacs > parses zodiachelp.txt correctly (snapshot) 1`] = `
{
  "Aquarius": {
    "info": "Best/Worst - Leo; Good - Gemini, Libra; Bad - Taurus, Scorpio
",
    "name": "Aquarius",
  },
  "Aries": {
    "info": "Best/Worst - Libra; Good - Sagittarius, Leo; Bad - Cancer, Capricorn
",
    "name": "Aries",
  },
  "Cancer": {
    "info": "Best/Worst - Capricorn; Good - Scorpio, Leo; Bad - Libra, Aries
",
    "name": "Cancer",
  },
  "Capricorn": {
    "info": "Best/Worst - Cancer; Good - Taurus, Virgo; Bad - Libra, Aries
",
    "name": "Capricorn",
  },
  "Gemini": {
    "info": "Best/Worst - Sagittarius; Good - Libra, Aquarius; Bad - Virgo, Pisces
",
    "name": "Gemini",
  },
  "Leo": {
    "info": "Best/Worst - Aquarius; Good - Sagittarius, Aries; Bad - Scorpio, Taurus
",
    "name": "Leo",
  },
  "Libra": {
    "info": "Best/Worst - Aries; Good - Aquarius, Gemini; Bad - Capricorn, Cancer
",
    "name": "Libra",
  },
  "Pisces": {
    "info": "Best/Worst - Virgo; Good - Cancer, Scorpio; Bad - Sagittarius, Gemini
",
    "name": "Pisces",
  },
  "Sagittarius": {
    "info": "Best/Worst - Gemini; Good - Aries, Leo; Bad - Pisces, Virgo
",
    "name": "Sagittarius",
  },
  "Scorpio": {
    "info": "Best/Worst - Taurus; Good - Pisces, Cancer; Bad - Aquarius, Leo
",
    "name": "Scorpio",
  },
  "Serpentarius": {
    "info": "Neutral to all",
    "name": "Serpentarius",
  },
  "Taurus": {
    "info": "Best/Worst - Scorpio; Good - Virgo, Capricorn; Bad - Leo, Aquarius
",
    "name": "Taurus",
  },
  "Virgo": {
    "info": "Best/Worst - Pisces; Good - Capricorn, Taurus; Bad - Sagittarius, Gemini
",
    "name": "Virgo",
  },
}
`;
